### PR TITLE
Ensure exact match links are only checked on paths containing separators 

### LINKED ISSUE #153832: File links without folders when command detection should prefer searching over matching a file in the initial cwd
With file structure on Windows[^1]:

- package.json
- src/
  - package.json

When shell integration is disabled, clicking a `package.json` link currently open `./package.json` directly.

The expectation here because cwd detection isn't possible on Windows unless shell integration is enabled it to prefer a search since we cannot tell what the active directory is. Consider the following problem case:

1. Open terminal
2. `cd src`
3. `ls`
4. Click `package.json`, 🐛 `./package.json` will open but a search should as the cwd is ambiguous

[^1]: The cwd can be detected on Linux/macOS even if shell integration is disabled

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes #153832

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/links/terminalLinkOpeners.ts [hunk 1]
```diff
@@ -218,6 +218,15 @@ export class TerminalSearchLinkOpener implements ITerminalLinkOpener {
 
 	private async _tryOpenExactLink(text: string, link: ITerminalSimpleLink): Promise<boolean> {
 		const sanitizedLink = text.replace(/:\d+(:\d+)?$/, '');
+		// For only file links disallow exact link matching, for example searching for `foo.txt`
+		// when no cwd information is available should search when only the initial cwd is available
+		// as it's ambiguous if there are multiple matches.
+		//
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/links/terminalLinkOpeners.ts:224]: this is confusing to me

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/links/terminalLinkOpeners.ts:224]: so when no cwd is available, search using initialCwd?

**Review by meganrogge — CHANGES_REQUESTED** [review]
a bit confused by the comment

File: src/vs/workbench/contrib/terminal/browser/links/terminalLinkOpeners.ts (hunk 1 above)
Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/links/terminalLinkOpeners.ts]: Yeah that was a bit rough lol, what do you think of the updated one?

**Review by meganrogge — APPROVED** [review]
👍🏼

**Pull request merged**